### ISSUE TITLE
getTemplateGroup does not return any templates

### ISSUE THREAD

**Issue by fritzmg** [contao/contao#682]
**Affected version(s)**
4.8.1, 4.8.x-dev (4070c7a07b)

**Description**  
As mentioned [here](https://github.com/contao/contao/pull/657#issuecomment-525814175) `Controller::getTemplateGroup` does not return any templates anymore, if the requested template group has the structure `foo_bar_` and the template names are `foo_bar_lorem`, `foo_bar_ipsum`, ...

**How to reproduce** 
You can quickly test this by altering the `news-bundle`:

1. In `tl_module.php` of the `news-bundle`, change the requested template group in `tl_module_news::getNewsTemplates` to `news_foo_`.
2. In the templates folder of the `news-bundle`, change the name of the news templates to `news_foo_*`.
3. Rebuild the application cache.
4. Create or edit a news list module and go to the template section.

You will not be able to select a `news_foo_*` template.

**Comment by leofeyer** [comment]
That is because `$blnSeparateOthers` is enabled automatically if there is more than one underscore:

https://github.com/contao/contao/blob/4070c7a07b434e542b59d63d1a5b696c2e45ff4e/core-bundle/src/Resources/contao/library/Contao/Controller.php#L91

I guess we cannot rely on this simple logic, therefore we need a second argument to explicitly enable the new behavior. When `foo_bar`, `foo_bar_lorem` and `foo_bar_ipsum` are in `vendor/contao/news-bundle/src/Resources/contao/templates`:

```php
Controller::getTemplateGroup('foo_bar_');
// will return ['foo_bar_lorem', 'foo_bar_ipsum']

Controller::getTemplateGroup('foo_bar');
// will return ['foo_bar', 'foo_bar_lorem', 'foo_bar_ipsum']

Controller::getTemplateGroup('foo_bar_', true);
// will return []

Controller::getTemplateGroup('foo_bar', true);
// will return ['foo_bar']
```

When only `foo_bar` is in the news bundle folder and the other two are in the `/templates` folder:

```php
Controller::getTemplateGroup('foo_bar_');
// will return ['foo_bar_lorem', 'foo_bar_ipsum']

Controller::getTemplateGroup('foo_bar');
// will return ['foo_bar', 'foo_bar_lorem', 'foo_bar_ipsum']

Controller::getTemplateGroup('foo_bar_', true);
// will return ['foo_bar_lorem', 'foo_bar_ipsum']

Controller::getTemplateGroup('foo_bar', true);
// will return ['foo_bar', 'foo_bar_lorem', 'foo_bar_ipsum']
```

**Comment by leofeyer** [comment]
Fixed in 090578be9fc41f5f6bc5f31a3233221a2c5032ed.
